update pull request template (#5545)
open-telemetry/opentelemetry.io: .github/PULL_REQUEST_TEMPLATE.md

<!--

> Please provide a meaningful description of what this change will do, or is for. Bonus points for including links to related issues, other PRs, or technical references.

Reviewer: ```suggestion
Please provide a meaningful description of what your changes will do. Bonus points for including links to related issues, other PRs, or technical references.
```

> By including a description, help help reviewers to understand the context of your change, which will reduce
> the time to review.

Reviewer: ```suggestion
A detailed description helps reviewers understand the context of your change and reduces
the time needed to review.
```

Please make sure to follow the [contribution guidelines](https://opentelemetry.io/docs/contributing/).

> We will run some checks on your PR. If they fail, take a look in to our [guide on pr-checks](https://opentelemetry.io/docs/contributing/pr-checks/).
> Most issues can be resolved by running `npm run fix:all` locally and pushing the changes.
> But please note that it is not your responsibility to fix all the issues. We will help you with that.

Reviewer: ```suggestion
We run checks on all PRs. If a check fails on your PR, look for the fix in our [guide](https://opentelemetry.io/docs/contributing/pr-checks/).
Most issues can be resolved by running `npm run fix:all` locally and pushing the changes.
But please note that it is not your responsibility to fix these issues. We will help you.
```
